support-models/src/main/scala/com/gu/support/workers/PaymentFields.scala

```diff
@@ -45,6 +67,15 @@ object PaymentFields {
       Decoder[StripePaymentMethodPaymentFields].widen,
       Decoder[DirectDebitPaymentFields].widen,
       Decoder[ExistingPaymentFields].widen
-    ).reduceLeft(_ or _)
+    ).reduceLeft(or(_,_))
```

Comment:
the alternaitve OR was needed becuase the built in one discards all but the last error.  In our case that's pointless as most of the time we don't expect the last one to succeed.
The implementation I've done is not great, but this is the annoying thing about non deterministic code.
Given more motivation, I would get it to serialising the subtype as a separate data point, as per withDiscriminator or similar https://circe.github.io/circe/codecs/adt.html#the-future and then it takes the non determinism out of it